submarine-workbench/workbench-server/src/test/java/org/apache/submarine/server/AbstractWorkbenchServerTest.java

```diff
@@ -0,0 +1,304 @@
+      url = System.getProperty("url");
+    } else {
+      url = "http://localhost:8080";
+    }
+    if (path != null) {
+      url += path;
+    }
+    return url;
+  }
+
+  public static String getWebsocketApiUrlToTest() {
+    String websocketUrl = "ws://localhost:8080" + WEBSOCKET_API_URL;
+    if (System.getProperty("websocketUrl") != null) {
+      websocketUrl = System.getProperty("websocketurl");
+    }
+    return websocketUrl;
+  }
+
+  public static String getUrlToTest() {
+    String url = "http://localhost:8080";
+    if (System.getProperty("url") != null) {
+      url = System.getProperty("url");
+    }
+    return url;
+  }
+
+  static ExecutorService executor;
+  protected static final Runnable SERVER = new Runnable() {
+    @Override
+    public void run() {
+      try {
+        TestUtils.clearInstances();
+        WorkbenchServer.main(new String[]{""});
+      } catch (Exception e) {
+        LOG.error("Exception in WebDriverManager while getWebDriver ", e);
+        throw new RuntimeException(e);
+      }
+    }
+  };
+
+  private static void start(boolean withAuth, String testClassName,
+      boolean withKnox, boolean cleanData) throws Exception {
+    LOG.info("Starting WorkbenchServer withAuth: {}, testClassName: {}, withKnox: {}",
+        withAuth, testClassName, withKnox);
+
+    if (!WAS_RUNNING) {
+      // copy the resources files to a temp folder
+      workbenchServerHome = new File("..");
+      LOG.info("SUBMARINE_WORKBENCH_SERVER_HOME: "
+          + workbenchServerHome.getAbsolutePath());
+      confDir = new File(workbenchServerHome, "conf_" + testClassName);
+      confDir.mkdirs();
+
+      System.setProperty(SubmarineConfiguration.ConfVars.WORKBENCH_WEB_WAR.getVarName(),
+          new File("../workbench-web/dist").getAbsolutePath());
+      System.setProperty(SubmarineConfiguration.ConfVars.SUBMARINE_CONF_DIR.getVarName(),
+          confDir.getAbsolutePath());
+      notebookDir = new File(workbenchServerHome.getAbsolutePath() + "/notebook_" + testClassName);
+      if (cleanData) {
+        FileUtils.deleteDirectory(notebookDir);
+      }
+
+      // some test profile does not build zeppelin-web.
+      // to prevent zeppelin starting up fail, create zeppelin-web/dist directory
+      new File("../zeppelin-web/dist").mkdirs();
+
+      LOG.info("Staring test Workbench server up...");
+
+      executor = Executors.newSingleThreadExecutor();
+      executor.submit(SERVER);
+      long s = System.currentTimeMillis();
+      boolean started = false;
+      while (System.currentTimeMillis() - s < 1000 * 60 * 3) {  // 3 minutes
+        Thread.sleep(2000);
+        started = checkIfServerIsRunning();
+        if (started == true) {
+          break;
+        }
+      }
+      if (started == false) {
+        throw new RuntimeException("Can not start Zeppelin server");
+      }
+      //ZeppelinServer.notebook.setParagraphJobListener(NotebookServer.getInstance());
```

Comment:
Need change to `//SubmarineServer.notebook.setParagraphJobListener(NotebookServer.getInstance());` ?

Reply:
Done.